tests/src/test/spark350/scala/com/nvidia/spark/rapids/iceberg/GpuIcebergPartitionerSuite.scala

```diff
@@ -0,0 +1,126 @@
+import com.nvidia.spark.rapids.iceberg.GpuIcebergPartitionerSuite.assertEqual
+import com.nvidia.spark.rapids.spill.SpillFramework
+import org.apache.iceberg.{PartitionKey, PartitionSpec, Schema, StructLike}
+import org.apache.iceberg.spark.{GpuTypeToSparkType, SparkStructLike}
+import org.apache.iceberg.types.Types
+import org.apache.iceberg.types.Types.StructType
+import org.scalatest.Assertions.assertResult
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.funsuite.AnyFunSuite
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
+
+class GpuIcebergPartitionerSuite extends AnyFunSuite with BeforeAndAfterAll {
+  private var seed = 0L
+
+  override def beforeAll: Unit = {
+    SpillFramework.initialize(new RapidsConf(new SparkConf))
+    seed = System.currentTimeMillis()
+    println(s"Random seed set to $seed")
+  }
+
+  override def afterAll: Unit = {
+    SpillFramework.shutdown()
+  }
+
+  test("bucket") {
+    val icebergSchema = new Schema(
+      Types.NestedField.required(1, "x", Types.LongType.get),
+      Types.NestedField.required(2, "y", Types.LongType.get, "comment"),
+      Types.NestedField.required(3, "z", Types.LongType.get),
+      Types.NestedField.required(4, "d", Types.IntegerType.get),
+    )
+
+    val sparkType = GpuTypeToSparkType.toSparkType(icebergSchema)
+
+    val icebergPartitionSpec = PartitionSpec.builderFor(icebergSchema)
+        .bucket("x", 16)
+        .bucket("d", 20)
+        .build()
+
+    val partitioner = new GpuIcebergPartitioner(icebergPartitionSpec, sparkType)
+
+    val partitioned = partitioner.partition(createColumnarBatch(sparkType, 1000, seed = seed))
+
+    val icebergPartitionKey = new PartitionKey(icebergPartitionSpec, icebergSchema)
+    val icebergRow = new SparkStructLike(icebergSchema.asStruct())
+
+    withResource(partitioned) { _ =>
+      for ((part, partIdx) <- partitioned.zipWithIndex) {
+        withResource(part.batch.getColumnarBatch()) { batch =>
+          assert(batch.numRows() > 0, s"partition $partIdx is empty")
+          val numCols = batch.numCols()
+          val hostCols = closeOnExcept(new Array[ColumnVector](numCols)) { hostCols =>
+            for (i <- 0 until numCols) {
+              hostCols(i) = batch.column(i).asInstanceOf[GpuColumnVector].copyToHost()
+            }
+            hostCols
+          }
+
+          withResource(new ColumnarBatch(hostCols, batch.numRows())) { hostBatch =>
+            for (rowIdx <- 0 until hostBatch.numRows()) {
+              val internalRow = hostBatch.getRow(rowIdx)
+              val row = Row.fromSeq(internalRow.toSeq(sparkType))
+              icebergRow.wrap(row)
+
+              icebergPartitionKey.partition(icebergRow)
+              val expectedPartitionKey = icebergPartitionKey
+              val actualPartitionKey = part.partition
+
+              assertEqual(expectedPartitionKey, actualPartitionKey,
```

Comment:
NIT: also assert the total row number of the result groups is equal to original input data row number, and more collect a map on CPU to assert each group is equal betwen CPU and GPU.